File: advanced-se6/src/main/java/com/turing/advancedse6/oop/InterfaceDemo.java
package com.turing.advancedse6.oop;

interface Flyable
{
	void fly();
}
class Bird implements Flyable
{
	public void fly()
	{
		System.out.println("Bird flying");
	}
}
class Aeroplane implements Flyable
{
	public void fly()
	{
		System.out.println("Aeroplane flying");
	}
}
public class InterfaceDemo {
	public static void main(String[]args)
	{
		Flyable flyable = new Bird();
		flyable.fly();
		
		flyable = new Aeroplane();
		flyable.fly();
	}
}

File: code/interface_demo.go
package main

import "fmt"

type Flyable interface {
	fly()
}
type Bird struct {
}

func (b Bird) fly() {
	fmt.Println("Bird Fly")
}

type Aeroplane struct {
}

func (a Aeroplane) fly() {
	fmt.Println("Aeroplane Fly")
}
func main() {
	var b Bird
	var a Aeroplane
	var flyable Flyable
	flyable = b
	flyable.fly()

	flyable = a
	flyable.fly()
}
